src/main/kotlin/no/nav/medlemskap/domene/Oppholdstillatelse.kt

```diff
@@ -10,13 +11,39 @@ data class Oppholdstillatelse(
     val arbeidsadgang: Arbeidsadgang?,
     val uavklartFlyktningstatus: Boolean?,
     val harFlyktningstatus: Boolean?
-)
+) {
+    fun harGyldigOppholdOgArbeidstillatelse(): Boolean {
+        return gjeldendeOppholdsstatus != null && gjeldendeOppholdsstatus.harTillatelse &&
+            arbeidsadgang != null && arbeidsadgang.harArbeidsadgang &&
+            arbeidsadgang.arbeidsomfang != null && arbeidsadgang.arbeidsomfang == ArbeidomfangKategori.KUN_ARBEID_HELTID &&
+            arbeidsadgang.arbeidsadgangType != null && arbeidsadgang.arbeidsadgangType == ArbeidsadgangType.GENERELL
+    }
+
+    fun harGyldigOppholdOgArbeidstillatelseForPeriode(
+        inputPeriode: InputPeriode,
+        førsteDagForYtelse: LocalDate?
+    ): Boolean {
+        return harGyldigOppholdOgArbeidstillatelse() &&
+            gjeldendeOppholdsstatus!!.periode.encloses(
+                Kontrollperiode.kontrollPeriodeForOppholdstillatelse(
+                    inputPeriode,
+                    førsteDagForYtelse
+                ).periode
+            ) &&
+            arbeidsadgang!!.periode.encloses(
+                Kontrollperiode.kontrollPeriodeForOppholdstillatelse(
+                    inputPeriode,
+                    førsteDagForYtelse
+                ).periode
+            )
+    }
+}
```

Comment:
Bør egentlig ikke dette vær i regelkjøringen?

Reply:
Den brukes i regel 15: GyldigOppholdstillatelseIKontrollperiodeRegel linje 29. Planen er å bruke den i regel 16 også, men det krever litt omskriving av type (periode vs inputperiode), så tar hele regel 16 i egen PR for å begrense denne PR'n.

Reply:
Skjønner, det ser fint ut :D Bare tenker på at det er veldig mye inni GjeldendeOppholdstatus som kanskje burde vært med i datagrunnlaget på sikt, men det kan vi ta da.